### ISSUE TITLE
Jet server breaks IMDG client doing MultiAttributeProjection

### ISSUE THREAD

**Issue by neilstevenson** [hazelcast/hazelcast#15181]
Jet defines -302 for `Object[]` - https://github.com/hazelcast/hazelcast-jet/blob/master/hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/serialization/SerializerHookConstants.java#L32

If a cluster is upgraded from an IMDG cluster to an IMDG + Jet cluster, then pure IMDG clients hit issues deserializing `Object[]`. This is the case for `MultiAttributeProjection`.

The simple answer is obviously to change the client from IMDG to IMDG + Jet, but this means retesting for the user.

```
Caused by: com.hazelcast.nio.serialization.HazelcastSerializationException: There is no suitable de-serializer for type -302. This exception is likely to be caused by differences in the serialization configuration between members or between clients and members.
	at com.hazelcast.internal.serialization.impl.AbstractSerializationService.newHazelcastSerializationException(AbstractSerializationService.java:238)
	at com.hazelcast.internal.serialization.impl.AbstractSerializationService.toObject(AbstractSerializationService.java:182)
	at com.hazelcast.spi.impl.UnmodifiableLazyList$UnmodifiableLazyListIterator.deserializeAndSet(UnmodifiableLazyList.java:167)
	at com.hazelcast.spi.impl.UnmodifiableLazyList$UnmodifiableLazyListIterator.next(UnmodifiableLazyList.java:142)
```

I have a (large) reproducer if need be.

**Comment by neilstevenson** [comment]
See https://github.com/hazelcast/hazelcast/issues/15179

**Comment by sancar** [comment]
@neilstevenson 
Is this duplicate of https://github.com/hazelcast/hazelcast-jet/issues/1444 ? 
If yes, I will close this since it is fixed via hazelcast/hazelcast#15506

**Comment by neilstevenson** [comment]
@sancar  Yes it's a duplicate
